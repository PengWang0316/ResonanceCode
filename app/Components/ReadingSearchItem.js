import React from 'react';
import PropTypes from 'prop-types';

import styles from '../styles/ReadingSearchItem.module.css';

const ReadingSearchItem = ({
  readingName, readingId, pingPongStates,
  readingIndex, handleDeleteCallback, handlePingPongStateChangeCallback
}) => {
  const handleDelete = _ => handleDeleteCallback(readingId, readingIndex);
  const handlePingPongStateChange = ({ target }) =>
    handlePingPongStateChangeCallback(readingId, target.value);
  return (
    <div>
      <div className={`row ${styles.readingListNameDiv}`}>
        <div className="col-xs-10">{readingName}</div>
        <div role="button" tabIndex="-2" className="col-xs-2 pr-1" onClick={handleDelete}><i className={`fa fa-trash ${styles.deleteIcon}`} /></div>
      </div>

      <div className="row pl-2">
        <div className="col-xs-1">&#8627;</div>
        <div className="col-xs-10">
          <select id="pingPongState" className="form-control" defaultValue={pingPongStates[readingId] ? pingPongStates[readingId] : 'Pre-reading'} onChange={handlePingPongStateChange}>
            <option value="Pre-reading">Pre-reading</option>
            <option value="Inquiring">Inquiring</option>
            <option value="Listening">Listening</option>
            <option value="Mystery Speaking">Field Speaking</option>
            <option value="Integrating Information">Integrating Information</option>
            <option value="Responding">Responding</option>
            <option value="Completion">Completion</option>
          </select>
        </div>
      </div>
    </div>
  );
};
ReadingSearchItem.propTypes = {
  readingName: PropTypes.string.isRequired,
  readingId: PropTypes.string.isRequired,
  pingPongStates: PropTypes.object.isRequired,
  readingIndex: PropTypes.number.isRequired,
  handleDeleteCallback: PropTypes.func.isRequired,
  handlePingPongStateChangeCallback: PropTypes.func.isRequired
};
export default ReadingSearchItem;
