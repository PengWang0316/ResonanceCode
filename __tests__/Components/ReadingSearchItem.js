import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import ReadingSearchItem from '../../app/Components/ReadingSearchItem';

describe('ReadingSearchItem test', () => {
  const defaultProps = {
    readingName: 'readingName',
    readingId: 'readingId',
    pingPongStates: {},
    readingIndex: 0,
    handleDeleteCallback: jest.fn(),
    handlePingPongStateChangeCallback: jest.fn()
  };
  const getShallowComponent = (props = defaultProps) => shallow(<ReadingSearchItem {...props} />);

  test('ReadingSearchItem snapshot', () => expect(renderer.create(<ReadingSearchItem {...defaultProps} />).toJSON()).toMatchSnapshot());

  test('handleDelete', () => {
    const component = getShallowComponent();
    component.find({ role: 'button' }).simulate('click');
    expect(defaultProps.handleDeleteCallback)
      .toHaveBeenLastCalledWith(defaultProps.readingId, defaultProps.readingIndex);
  });

  test('handlePingPongStateChange', () => {
    const component = getShallowComponent({ ...defaultProps, pingPongStates: { readingId: 'pingPongState' } });
    component.find('select').simulate('change', { target: { value: 'newValue' } });
    expect(defaultProps.handlePingPongStateChangeCallback).toHaveBeenLastCalledWith(defaultProps.readingId, 'newValue');
  });
});
