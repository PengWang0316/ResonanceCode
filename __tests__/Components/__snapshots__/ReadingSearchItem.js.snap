// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadingSearchItem test ReadingSearchItem snapshot 1`] = `
<div>
  <div
    className="row readingListNameDiv"
  >
    <div
      className="col-xs-10"
    >
      readingName
    </div>
    <div
      className="col-xs-2 pr-1"
      onClick={[Function]}
      role="button"
      tabIndex="-2"
    >
      <i
        className="fa fa-trash deleteIcon"
      />
    </div>
  </div>
  <div
    className="row pl-2"
  >
    <div
      className="col-xs-1"
    >
      â†³
    </div>
    <div
      className="col-xs-10"
    >
      <select
        className="form-control"
        defaultValue="Pre-reading"
        id="pingPongState"
        onChange={[Function]}
      >
        <option
          value="Pre-reading"
        >
          Pre-reading
        </option>
        <option
          value="Inquiring"
        >
          Inquiring
        </option>
        <option
          value="Listening"
        >
          Listening
        </option>
        <option
          value="Mystery Speaking"
        >
          Field Speaking
        </option>
        <option
          value="Integrating Information"
        >
          Integrating Information
        </option>
        <option
          value="Responding"
        >
          Responding
        </option>
        <option
          value="Completion"
        >
          Completion
        </option>
      </select>
    </div>
  </div>
</div>
`;
